pyro/primitives.py

```diff
@@ -61,6 +63,30 @@ def param(name, *args, **kwargs):
     return _param(name, *args, **kwargs)
 
 
+def _masked_observe(name, fn, obs, obs_mask, *args, **kwargs):
+    # Split into two auxiliary sample sites.
+    with poutine.mask(mask=obs_mask):
+        observed = sample(f"{name}_observed", fn, *args, **kwargs, obs=obs)
+    with poutine.mask(mask=~obs_mask):
+        unobserved = sample(f"{name}_unobserved", fn, *args, **kwargs)
```

Comment:
cool so this will let downstream sample statements depend on the sampled `unobserved` values right (since they will appear in `value` together with `observed`)?

Reply:
Correct. And it took less than three years to figure out how to do it 😄